akka-docs/rst/java/code/docs/http/javadsl/server/directives/DebuggingDirectivesExamplesTest.java

```diff
@@ -127,5 +127,52 @@ public void testLogResult() {
       .assertEntity("logged");
     //#logResult
   }
+   @Test
+  public void testLogRequestResultWithResponseTime() {
+    //#logRequestResultWithResponseTime
+    // using logRequestResultOptional for generating Response Time
 
-}
\ No newline at end of file
+    // handle request to optionally generate a log entry
+
+    BiFunction<HttpRequest, HttpResponse , Optional<LogEntry>> requestMethodAsInfo=
+     (request, response) ->{
+     Long requestTime=System.nanoTime();
+      return printResponseTime(request,response,requestTime);
+	};
+     
+    // handle rejections to optionally generate a log entry
+    BiFunction<HttpRequest, List<Rejection>, Optional<LogEntry>> rejectionsAsInfo = 
+      (request, rejections) ->
+        (!rejections.isEmpty())  ? 
+          Optional.of(
+            LogEntry.create(
+              rejections
+                .stream()
+                .map(Rejection::toString)
+                .collect(Collectors.joining(", ")), 
+              InfoLevel()))
+          : Optional.empty(); // no rejections
+
+    final Route route = get(() -> logRequestResultOptional(
+            requestMethodAsInfo,
+      rejectionsAsInfo,
+      () -> complete("logged")));
+    // tests:
+    testRoute(route).run(HttpRequest.GET("/")).assertEntity("logged");
+    //#logRequestResult
+  }
+ // A function for the logging of Time
+  public static Optional<LogEntry> printResponseTime(HttpRequest request, HttpResponse response, Long requestTime){
+    if(response.status().isSuccess())
+    {
+      Long elapsedTime=(requestTime - System.nanoTime()) / 1000;
+       return Optional.of(
+              LogEntry.create(
+                      "Logged Request:"+request.method().name()+":"+request.getUri()+":"+response.status()+":"+elapsedTime,
+                      InfoLevel()));
+    }
+    else{
+     return Optional.empty();//not a successfull response
+    }
+  }
+}
```

Comment:
For the identation I use Intellij and when I try to ident it , it changes a lot of lines hence I tried to do it manually. What is off here ? Can you please point it out so that I may do it manually.